prelude.dx

```diff
@@ -581,3 +588,83 @@ def (<>) (x:List a) (y:List a) : List a =
     case i' < nx of
       True  -> xs.(fromOrdinal _ i')
       False -> ys.(fromOrdinal _ (i' - nx))
+
+'Isomorphisms
+
+data Iso a:Type b:Type = MkIso { fwd: a -> b & bwd: b -> a }
+
+def appIso (iso: Iso a b) (x:a) : b =
+  (MkIso {fwd, bwd}) = iso
+  fwd x
+
+def flipIso (iso: Iso a b) : Iso b a =
+  (MkIso {fwd, bwd}) = iso
+  MkIso {fwd=bwd, bwd=fwd}
+
+def revIso (iso: Iso a b) (x:b) : a = appIso (flipIso iso) x
+
+idIso : Iso a a = MkIso {fwd=id, bwd=id}
+
+def (<>>) (iso1: Iso a b) (iso2: Iso b c) : Iso a c =
+  (MkIso {fwd=fwd1, bwd=bwd1}) = iso1
+  (MkIso {fwd=fwd2, bwd=bwd2}) = iso2
+  MkIso {fwd=(fwd1 >>> fwd2), bwd=(bwd1 <<< bwd2)}
+
+def (<<>) (iso2: Iso b c) (iso1: Iso a b) : Iso a c = iso1 <>> iso2
```

Comment:
I find those operators _really_ hard to read. I feel like it would be simpler to pick a consistent leading char and make them of the form `x>>` and `x<<`. I would probably even stomach making them 4 char long instead of mixing up different directions in a single operator.

Reply:
That's a fair point. I kinda liked how an isomorphism is bidirectional and so one of the arrows points the other way, but I did have to look at it a few times to make sure I was doing the direction right.

Perhaps `<<&` and `&>>`?